coramin/third_party/minlplib_tools.py

```diff
@@ -0,0 +1,336 @@
+
+    Parameters
+    ----------
+    download_dir: str
+        The directory in which to place the downloaded file. The default will be a 
+        directory called minlplib in the current working directory.
+    """
+    if download_dir is None:
+        download_dir = os.path.join(os.getcwd(), 'minlplib')
+
+    if not os.path.exists(download_dir):
+        os.makedirs(download_dir)
+
+    downloader = download.FileDownloader()
+    downloader.set_destination_filename(os.path.join(download_dir, 'instancedata.csv'))
+    downloader.get_binary_file('http://www.minlplib.org/instancedata.csv')
+
+
+def filter_minlplib_instances(instancedata_download_dir=None,
+                              min_nvars=0, max_nvars=math.inf,
+                              min_nbinvars=0, max_nbinvars=math.inf,
+                              min_nintvars=0, max_nintvars=math.inf,
+                              min_nnlvars=0, max_nnlvars=math.inf,
+                              min_nnlbinvars=0, max_nnlbinvars=math.inf,
+                              min_nnlintvars=0, max_nnlintvars=math.inf,
+                              min_nobjnz=0, max_nobjnz=math.inf,
+                              min_nobjnlnz=0, max_nobjnlnz=math.inf,
+                              min_ncons=0, max_ncons=math.inf,
+                              min_nlincons=0, max_nlincons=math.inf,
+                              min_nquadcons=0, max_nquadcons=math.inf,
+                              min_npolynomcons=0, max_npolynomcons=math.inf,
+                              min_nsignomcons=0, max_nsignomcons=math.inf,
+                              min_ngennlcons=0, max_ngennlcons=math.inf,
+                              min_njacobiannz=0, max_njacobiannz=math.inf,
+                              min_njacobiannlnz=0, max_njacobiannlnz=math.inf,
+                              min_nlaghessiannz=0, max_nlaghessiannz=math.inf,
+                              min_nlaghessiandiagnz=0, max_nlaghessiandiagnz=math.inf,
+                              min_nsemi=0, max_nsemi=math.inf,
+                              min_nnlsemi=0, max_nnlsemi=math.inf,
+                              min_nsos1=0, max_nsos1=math.inf,
+                              min_nsos2=0, max_nsos2=math.inf,
+                              acceptable_formats=None,
+                              acceptable_probtype=None,
+                              acceptable_objtype=None,
+                              acceptable_objcurvature=None,
+                              acceptable_conscurvature=None,
+                              acceptable_convex=None):
+    """
+    This function filters problems from MINLPLib based on
+    instancedata.csv from MINLPLib and the conditions specified
+    through the function arguments. The function argument names
+    correspond to column headings from instancedata.csv. The 
+    arguments starting with min or max require integer inputs. 
```

Comment:
It's interesting that you are requiring int in the documentation, as the default value for the max is a float().

Reply:
I agree it is slightly weird. However, I'm not sure I have a better approach. It would make sense to use an int version of inf, but that does not exist.

Reply:
Why not just document that it accepts float?  Logically, there is nothing wrong with a floating point upper bound for an integer variable (unless you ever need to test that the bound is binding...)

Reply:
Good point...

Reply:
Fixed